api/envoy/config/rbac/v2alpha/rbac.proto

```diff
@@ -60,96 +67,76 @@ message RBAC {
-      // The map key is the header name. The header specifies how the service is accessed.
-      MapEntryMatch header = 1;
-
-      // Destination IP addresses.
-      IpMatch destination_ips = 2;
-
-      // Destination ports.
-      PortMatch destination_ports = 3;
-    }
-  }
+    // A set of rules where at least one must match in order to define the action.
+    Set or_rules = 2;
 
-  // Optional. Custom conditions.
-  repeated Condition conditions = 3;
+    // When any is set, it matches any action.
+    bool any = 3 [(validate.rules).bool.const = true];
+
+    // A header (or psuedo-header such as :path or :method) on the incoming HTTP request.
+    envoy.api.v2.route.HeaderMatcher header = 4;
+
+    // A CIDR block that describes the destination IP.
+    envoy.api.v2.core.CidrRange destination_ip = 5;
+
+    // A port number that describes the destination port connecting to.
+    uint32 destination_port = 6 [(validate.rules).uint32.lte = 65535];
+  }
 }
 
 // Principal defines an identity or a group of identities.
 message Principal {
+
+  message Set {
+    repeated Principal ids = 1 [(validate.rules).repeated.min_items = 1];
+  }
+
   // Authentication attributes for principal. These could be filled out inside RBAC filter.
   // Or if an authentication filter is used, they can be provided by the authentication filter.
   message Authenticated {
-    // Optional. The name of the principal. This matches to the "source.principal" field in
-    // ":ref: `AttributeContext <envoy_api_msg_service.auth.v2alpha.AttributeContext>`.
-    // If unset, it applies to any user.
+    // The name of the principal. If set, the URI SAN is used from the certificate, otherwise the
+    // subject field is used. If unset, it applies to any user that is authenticated.
     string name = 1;
   }
 
-  // Optional. Authenticated attributes that identify the principal.
-  Authenticated authenticated = 1;
+  oneof identifier {
+    option (validate.required) = true;
 
-  // Definition of a custom attribute to identify the principal.
-  message Attribute {
-    oneof attribute_spec {
-      // Source service name. This matches to the "source.service" field in
-      // ":ref: `AttributeContext <envoy_api_msg_service.auth.v2alpha.AttributeContext>`.
-      string service = 1;
+    // A set of identifiers that all must match in order to define the principal.
+    Set and_ids = 1;
 
-      // Source IP addresses.
-      IpMatch source_ips = 2;
+    // A set of identifiers that all must match in order to define the principal.
+    Set or_ids = 2;
 
-      // Header match. This matches to the "request.http.headers" field in
-      // ":ref: `AttributeContext <envoy_api_msg_service.auth.v2alpha.AttributeContext>`.
-      // The map "key" is the header name. The header identifies the client.
-      MapEntryMatch header = 3;
-    }
-  }
+    // When any is set, it matches any principal.
+    bool any = 3 [(validate.rules).bool.const = true];
```

Comment:
Same question here. If "any" is unset or set to false, and no principal is defined, what is the behavior?

Reply:
"any" cannot be set to false, the validation rule precludes it.

Reply:
Thanks for clarification.